tekdoc/main.go

```diff
@@ -17,15 +17,91 @@ limitations under the License.
 package main
 
 import (
-	"fmt"
+	"flag"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
+	"github.com/tektoncd/pipeline/pkg/client/clientset/versioned/scheme"
+	"io/ioutil"
+	"log"
+	"os"
+	"text/template"
+)
 
-	_ "github.com/tektoncd/pipeline/pkg/pod"
+var (
+	filename = flag.String("f", "", "Name of the file to parse")
 )
 
+type a struct {
+	Taskname  string
+	Input     []v1alpha1.ParamSpec
+	Resources []v1alpha1.TaskResource
+	Output    []v1alpha1.TaskResource
+}
+
 func main() {
-	fmt.Println("Hello this is STILL TekDoc")
 
 	// TODO: take a file path, parse it as YAML -> v1alpha1.Task
+	flag.Parse()
+	dat, err := ioutil.ReadFile(*filename)
+	if err != nil {
+		log.Fatalf("error reading file: %v", err)
+	}
+
+	var task v1alpha1.Task
+
+	if _, _, err := scheme.Codecs.UniversalDeserializer().Decode(dat, nil, &task); err != nil {
+		log.Fatal(err)
+	}
+
 	// TODO: use Go templates to print v1alpha1.Task as Markdown
-	// TODO: tests!
+	x := a{
+		Taskname: task.Name,
+	}
+	tmpl := template.Must(template.New("test").Parse(`# {{.Taskname}}
+## Install the Task
+kubectl apply -f https://raw.githubusercontent.com/tektoncd/catalog/master/{{.Taskname}}/{{.Taskname}}.yaml
+### Input:-
+`))
+	if err = tmpl.Execute(os.Stdout, x); err != nil {
+		log.Fatalf("error executing the template: %v", err)
+	}
+	if task.Spec.Inputs.Params != nil {
```

Comment:
This still runs the risk of panicking if task.Spec or task.Spec.Inputs are nil.

I think we want to have a check and then stop execution if task.Spec is nil (since then there would be nothing to do), and then check if task.Spec.Inputs is nil here.